addon/components/frost-list.js

```diff
@@ -114,6 +136,36 @@ export default Component.extend({
     })
   },
 
+  @readOnly
+  @computed('componentKeyNamesForTypes')
+  isAnyTypedItemExpansion (componentKeyNamesForTypes) {
+    const componentKeyNames = this.get('componentKeyNames')
+    const itemExpansionKeyName = get(componentKeyNames, 'itemExpansion')
+
+    for (var itemType in componentKeyNamesForTypes) {
```

Comment:
I think we forgot one case. Let's say that all the type in `componentKeyNamesForTypes` has no expansion but that there is a default expansion, we will probably have to consider this as a case where we will expect expansion.

The other case we didn't consider is it's not because there is an expansion defined that there will be an expansion. Let's take an example where you have type a and b. b has expansion but not a. You only have elements of type a in the list so you will show all the element shifted to the right for no reason.

Reply:
We might want to take in consideration the content of the list + the type to determine this.

Reply:
Did you look at the cases defined here?

Reply:
I understand what you mean about expecting a default expansion when items have item component defined but no item expansion component defined. I disagree though as I think we have to draw a clear differentiation for the cases where the consumer is providing the components for the typed item. Up until now we have said that if the item is typed and there is a component provided for that type, then we will not show the default expansion. I think we need to stick to that even for the case when there are no items with a specific expansion component provided for a type. 

For the case when we have only certain types of items and so the list shows the 35px shift, I agree that we should handle this scenario.